protocol/contracts/admin/VestingPayments.sol

```diff
@@ -0,0 +1,259 @@
+    function createSchedules(
+        VestingSchedule[] calldata _schedules 
+    ) external override onlyElevatedAccess {
+        if (_schedules.length == 0) { revert CommonEventsAndErrors.InvalidParam(); }
+        uint256 _length = _schedules.length;
+        VestingSchedule calldata _schedule;
+        for (uint256 i; i < _length; ++i) {
+            _schedule = _schedules[i];
+            // distributed and revoked should be default values
+            if (_schedule.distributed != 0) { revert CommonEventsAndErrors.InvalidParam(); }
+            if (_schedule.revoked) { revert CommonEventsAndErrors.InvalidParam(); }
+            if (_schedule.revokedReleasable != 0) { revert CommonEventsAndErrors.InvalidParam(); }
+            if (_schedule.recipient == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+            if (_schedule.start < block.timestamp) { revert CommonEventsAndErrors.InvalidParam(); }
+            if (_schedule.cliff <= _schedule.start) { revert CommonEventsAndErrors.InvalidParam(); }
+            // total duration must be greater than cliff
+            // duration must be more than difference from start to cliff
+            if (_schedule.duration <= _schedule.cliff - _schedule.start) { revert CommonEventsAndErrors.InvalidParam(); }
+            if (_schedule.amount == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+
+            bytes32 vestingId = computeNextVestingScheduleIdForHolder(_schedule.recipient);
+            schedules[vestingId] = _schedule;
+            totalVestedAndUnclaimed += _schedule.amount;
+            /// @dev very low possibility of duplicate vesting IDs generated but still checking. 
+            if (!_activeVestingIds.add(vestingId)) { revert InvalidScheduleId(); }
+            holdersVestingCount[_schedule.recipient] += 1;
+            emit ScheduleCreated(vestingId, _schedule.recipient, _schedule.start, _schedule.cliff, _schedule.duration, _schedule.amount);
+        }
+    }   
+
+    /// @inheritdoc IVestingPayments
+    function revokeVesting(bytes32 _vestingId) external override onlyElevatedAccess {
+        if (!isActiveVestingId(_vestingId)) { revert CommonEventsAndErrors.InvalidParam(); }
+        VestingSchedule storage _schedule = schedules[_vestingId];
+        if (block.timestamp >= _schedule.duration + _schedule.start) { revert FullyVested(); }
+        /// @dev No need to check if `_schedule.revoked` because id is removed from _activeVestingIds so it will fail `isActiveVestingId`
+
+        uint256 vestedNow = _getTotalVestedAtCurrentTime(_schedule);
+        uint256 unvested = _schedule.amount - vestedNow;
+        if (unvested > 0) {
+            totalVestedAndUnclaimed -= unvested;
+        }
+        // Claimable delta at revoke time
+        uint256 releasableNow = vestedNow - _schedule.distributed;
+        // Persist per-schedule claimable checkpoint
+        _schedule.revokedReleasable = releasableNow.encodeUInt128();
```

Comment:
Was a specific test case added to check multiple revoke's for the same user? It doesn't look like it, but that's the exact case this was intended to fix. So that should 💯 be added in (unless i just missed it in the existing tests)

Reply:
This was already in the existing test [previous commit](https://github.com/TempleDAO/temple/pull/1169/files#diff-5bd7314dad1cc21bbc5f3aa22b8b8b5d1a05d6175e4c4bd247b0c8b5a027a5e0R733-R739) as
```
function test_revokeVesting_invalid_id_after_revoke() public {
        test_revokeVesting_immediate_revoke_after_creation();
        // use previous index to match id
        bytes32 _id = vesting.computeVestingScheduleIdForAddressAndIndex(alice, 0);
        vm.expectRevert(abi.encodeWithSelector(CommonEventsAndErrors.InvalidParam.selector));
        vesting.revokeVesting(_id);
    }
```
Unless I misunderstood your point?